File: python_ruby/oo.py
class Hello:
    def __init__(self,thing):
        self.thing = thing

    def get(self):
        return self.thing

    def set(self,x):
        self.thing = x

class HelloThere(Hello):

    def concatenate(self,x):
        self.thing += x

hi = Hello("hi")
print hi.get()
hi.set("hello")
print hi.get()

hi_t = HelloThere("Hi")
hi_t.concatenate(" there")
print hi_t.get()

File: oo.rb
class Hello
  attr_accessor :thing
  def initialize(thing)
    @thing = thing
  end

  def get
    return @thing
  end

  def set(x)
    @thing = x
  end
end

class HelloThere < Hello
  def concatenate(x)
    @thing += x
  end
end


hi = Hello.new("hi")
puts hi.get
hi.set("hello")
puts hi.get

hi_t = HelloThere.new("Hi")
hi_t.concatenate(" there")
puts hi_t.get
